File: src/main/java/net/atos/client/bag/util/GebruiksdoelEnumAdapter.java
/*
 * SPDX-FileCopyrightText: 2023 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.bag.util;

import jakarta.json.bind.adapter.JsonbAdapter;

import nl.info.client.bag.model.generated.Gebruiksdoel;

public class GebruiksdoelEnumAdapter implements JsonbAdapter<Gebruiksdoel, String> {

    @Override
    public String adaptToJson(final Gebruiksdoel gebruiksdoel) {
        return gebruiksdoel.toString();
    }

    @Override
    public Gebruiksdoel adaptFromJson(final String json) {
        return Gebruiksdoel.fromValue(json);
    }
}

File: src/main/kotlin/nl/info/client/bag/util/GebruiksdoelEnumAdapter.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.bag.util

import jakarta.json.bind.adapter.JsonbAdapter
import nl.info.client.bag.model.generated.Gebruiksdoel

class GebruiksdoelEnumAdapter : JsonbAdapter<Gebruiksdoel, String> {
    override fun adaptToJson(gebruiksdoel: Gebruiksdoel): String = gebruiksdoel.toString()
    override fun adaptFromJson(json: String): Gebruiksdoel = Gebruiksdoel.fromValue(json)
}
